src/adapter/src/optimize/peek.rs

```diff
@@ -308,6 +322,36 @@ impl<'s> Optimize<LocalMirPlan<ResolvedLocal<'s>>> for Optimizer {
             );
         }
 
+        // Adding s3 sink if initiated from copy to.
+        if let Some(copy_to_context) = copy_to_context {
+            let mz_storage_types::connections::Connection::Aws(aws_connection) =
+                &copy_to_context.connection
+            else {
+                // Currently only s3 sinks are supported. It was already validated in planning that this
+                // is an aws connection.
+                unreachable!()
+            };
+            // Creating an S3 sink as currently only s3 sinks are supported. It might be possible in the future
+            // for COPY TO to write to different sinks, which should be set here depending upon the
+            // url scheme.
+            let sink_desc = ComputeSinkDesc {
+                from_desc: copy_to_context.desc.clone(),
+                from: self.select_id,
+                connection: ComputeSinkConnection::S3Oneshot(S3OneshotSinkConnection {
+                    aws_connection: aws_connection.clone(),
+                    prefix: copy_to_context.to.to_string(),
+                }),
+                with_snapshot: true,
+                up_to: Default::default(), // this will get updated  when the GlobalMirPlan is resolved with as_of below
+                // No `FORCE NOT NULL` for copy_to
+                non_null_assertions: vec![],
+                // No `REFRESH` for copy_to
+                refresh_schedule: None,
+            };
+            // TODO (mouli): should the same select_id be used here?
```

Comment:
Should be fine, based on that for SUBSCRIBE we use the same `transient_id` for both the view output and the sink too (i.e. the sink gets `transient_id` as input but is also exported under `transient_id`). Seems to work, although it _is_ a bit weird.

Reply:
ty, removed the comment.